### PR TITLE
Issue #194 Update README to improve installation guide.

### PR DESCRIPTION
**Pull request by bendiy**
For: https://github.com/plv8/plv8/issues/194

### CODE REVIEW

**Comment by bendiy** [comment]
TODO: https://github.com/plv8/plv8/issues/161

**Comment by JerrySievert** [comment]
thanks for your work here, but ultimately I'd like to see this information go into `plv8.md` under `doc`, as this gets generated into `PGXN` documentation as well:

http://pgxn.org/dist/plv8/doc/plv8.html

will reply inline as to what might work well with README.md, and what might work better with plv8.md.

File: README.md [hunk 1]
```diff
@@ -1,132 +1,212 @@
...
+plv8 is a shared library that provides a PostgreSQL procedural language powered
+by V8 JavaScript Engine.  With this program you can write in JavaScript your
+function that is callable from SQL.
+
+# Requirements:
+plv8 is tested with:
+
 - PG: version 9.2, 9.3, 9.4 and 9.5 (maybe older/newer are allowed)
-- V8: version 4.4 to 4.10
+- V8: version 4.4 to 4.10
```
Comment (JerrySievert) [inline README.md]: actually, up to 5.1 at this point.

File: README.md [hunk 2]
```diff
@@ -1,132 +1,212 @@
...
+Also all tools that PostgreSQL and V8 require to be built are required if you
+are building those from source.
+
+# Installing plv8
+
+## Build from source:
+```shell
+$ git clone git@github.com:plv8/plv8.git
+$ cd ./plv8
+$ git checkout tags/v1.5.4 -b 1.5.4
```
Comment (JerrySievert) [inline README.md]: I would rather not get people checking out branches unless they have very special requirements.  for instance, `1.4` is maintained strictly for Debian compatibility, as they work their way to a modern `v8`.  `1.5` is maintained for legacy at this point as well.

great pains are made to not break master, where development on modern `v8` is being done, and more modern OS's will either ship with a newer `libv8`, or will require a more recent version of `v8` in order to compile at all without hacks.

File: README.md [hunk 3]
```diff
@@ -1,132 +1,212 @@
...
+installed, you can "hide" 9.6 from `pg_config`. One way to do this in Ubuntu 
+is to move the 9.6 directory:
+TODO: (Is there a better way?)
+```shell
+$ mv /usr/lib/postgresql/9.6 /usr/lib/
+# Then build pv8.
+$ make static
+# Then move it back.
+$ mv /usr/lib/9.6 /usr/lib/postgresql/
+```
```
Comment (JerrySievert) [inline README.md]: I feel this whole section should exist in `plv8.md` - a link would suffice.

**Comment by bendiy** [comment]
@JerrySievert Thanks for the feedback. I'll make a few more tweaks to this next week.

File: README.md [hunk 4]
```diff
@@ -1,132 +1,210 @@
...
-
+A Procedural Language in JavaScript powered by V8
+=================================================
+
+plv8 is a shared library that provides a PostgreSQL procedural language powered
+by V8 JavaScript Engine.  With this program you can write in JavaScript your
+function that is callable from SQL.
+
+# Requirements:
+plv8 is tested with:
```
Comment (bendiy) [inline README.md]: @JerrySievert Do you know if the:
```
REQUIREMENT
-----------
```

Format is needed for the PGXN wiki or if I can use the Markdown `#` formatter for titles?
```
# Requirements:
```

File: README.md (hunk 2 above)
Comment (ruimarinho) [inline README.md]: @JerrySievert wdyt about releasing current master HEAD as 2.0.0-dev1? That way I could probably tag a docker postgres-plv8 version for more widespread usage.

**Comment by bendiy** [comment]
@JerrySievert I was able to get back to this PR and finish it. I've moved pretty much all of the README to the doc/plv8.md file and cleaned up the formatting.

**Comment by JerrySievert** [comment]
will review soon, but in response to @ruimarinho's now missing (GitHub!!!!!!) comment, I've been thinking about it for a while, and while I'd love to release 2.0 with a debugger, it might be best to wait for 2.1 with that - I'll work on doing some cleanup and getting 2.0 released soon.

**Comment by bendiy** [comment]
Missing comment is here:
https://github.com/plv8/plv8/pull/198#discussion_r96537283

**Comment by JerrySievert** [comment]
thanks @bendiy :) also just asked for a feature request to get a link to the branch instead of just a hover over ...

**Comment by JerrySievert** [comment]
@bendiy I'll give some notes inline, but after that, considering @ruimarinho's request, I'd love it if you'd put in a PR against the 1.5 branch as well.

**Comment by JerrySievert** [comment]
(otherwise, this is awesome)

File: doc/plv8.md [hunk 5]
```diff
@@ -1,50 +1,248 @@
...
+# e.g.
+$ apt-get install postgresql-9.1-plv8
+# OR up to
+$ apt-get install postgresql-9.6-plv8
+```
+
+### Redhat/CentOS:
+TODO - PL/v8 supports Redhat/CentOS. A Pull Request for installation steps is greatly appreciated.
+
+### MacOS:
```
Comment (JerrySievert) [inline doc/plv8.md]:
```
$ brew install plv8
```

or `make static` and `make install`

File: doc/plv8.md [hunk 6]
```diff
@@ -1,50 +1,248 @@
...
+```
+
+### Testing PL/v8 on a database:
+Below are some example queries to test if the extension is working:
+```sql
+  DO $$
+    plv8.elog(WARNING, 'plv8.version = ' + plv8.version); -- Will output the PL/v8 installed as a PostgreSQL `WARNING`.
+  $$ LANGUAGE plv8;
+```
+As of 2.0.0, there is a function to determine which version of PL/v8 you have
```
Comment (JerrySievert) [inline doc/plv8.md]: I really need to backport this (note for myself, not for you)

File: doc/plv8.md (hunk 5 above)
Comment (bendiy) [inline doc/plv8.md]: done

**Comment by bendiy** [comment]
@JerrySievert 
I can add a PR for the 1.5 branch. I'll do that once you approve and merge.

**Comment by JerrySievert** [comment]
merged - I'll do an update on checkout when I cut the 2.0 release (likely this weekend)

**Comment by JerrySievert** [comment]
btw, awesome, thanks much!

**Comment by bendiy** [comment]
No problem. Thanks for all your work on this project too.
`r1.5` branch PR is here:
https://github.com/plv8/plv8/pull/204

**Pull request merged**